scalacheck/src/main/scala/pathy/scalacheck/PathOf.scala

```diff
@@ -54,19 +54,14 @@ object PathOf {
     genRelFile[A] map (rootDir </> _)
 
   private def genRelFile[A: Arbitrary: Show]: Gen[RelFile[Sandboxed]] =
-    for {
-      d <- genRelDir[A]
-      s <- genSegment[A]
-    } yield d </> file(s)
+    sizeDistributed(genRelDir[A], genSegment[A])((d, s) => d </> file(s))
 
   private def genAbsDir[A: Arbitrary: Show]: Gen[AbsDir[Sandboxed]] =
     genRelDir[A] map (rootDir </> _)
 
   private def genRelDir[A: Arbitrary: Show]: Gen[RelDir[Sandboxed]] =
```

Comment:
`size` is a very new concept to me in scalacheck, but currently I believe that this generator as implemented will not return a path with 10 segments if the `size` parameter is set to 10. If the behavior for `Gen[List[A]]` is to generate a list of 10 elements if the `size` parameter is set to 10, then I would say the current behavior is not optimal.